dist/libphonenumber_italy.js

```diff
@@ -0,0 +1,1285 @@
+    }
+
+    function S(a, b, c, d) {
+        switch (b) {
+            case 0:
+                return "+" + a + c + d;
+            case 1:
+                return "+" + a + " " + c + d;
+            case 3:
+                return "tel:+" + a + "-" + c + d;
+            default:
+                return c + d
+        }
+    }
+
+    function Na(a, b, c) {
+        b = 0 == (p(b, 20) || []).length || 2 == c ? p(b, 19) || [] : p(b, 20) || [];
+        b = W(b, a);
+        return null == b ? a : Qa(a, b, c, void 0)
+    }
+
+    function W(a, b) {
+        for (var c, d = a.length, e = 0; e < d; ++e) {
+            c = a[e];
+            var f = c.g[3].h ? null != c.a[3] ? c.a[3].length : 0 : null != c.a[3] ? 1 : 0;
+            if (0 == f || 0 == b.search(r(c, 3, f - 1)))
+                if (f = new RegExp(r(c, 1)), O(f, b)) return c
+        }
+        return null
+    }
+
+    function Qa(a, b, c, d) {
+        var e = t(b, 2),
+            f = new RegExp(r(b, 1)),
+            g = t(b, 5),
+            h = "";
+        2 == c && null != d && 0 < d.length && 0 < g.length ? (b = g.replace(Ga, d), e = e.replace(Da, b), h = a.replace(f, e)) : (b = t(b, 4), h = 2 == c && null != b && 0 < b.length ? a.replace(f, e.replace(Da, b)) : a.replace(f, e));
+        3 == c && (h = h.replace(RegExp("^[-x\u2010-\u2015\u2212\u30fc\uff0d-\uff0f \u00a0\u00ad\u200b\u2060\u3000()\uff08\uff09\uff3b\uff3d.\\[\\]/~\u2053\u223c\uff5e]+"), ""), h = h.replace(RegExp("[-x\u2010-\u2015\u2212\u30fc\uff0d-\uff0f \u00a0\u00ad\u200b\u2060\u3000()\uff08\uff09\uff3b\uff3d.\\[\\]/~\u2053\u223c\uff5e]+",
+            "g"), "-"));
+        return h
+    }
+
+    function Ma(a, b, c) {
+        return null != a.a[3] && 0 != r(a, 3).length ? 3 == c ? ";ext=" + r(a, 3) : null != b.a[13] ? r(b, 13) + t(a, 3) : " ext. " + t(a, 3) : ""
+    }
+
+    function Ra(a, b) {
+        return X(a, r(b, 1)) ? X(a, r(b, 5)) ? 4 : X(a, r(b, 4)) ? 3 : X(a, r(b, 6)) ? 5 : X(a, r(b, 8)) ? 6 : X(a, r(b, 7)) ? 7 : X(a, r(b, 21)) ? 8 : X(a, r(b, 25)) ? 9 : X(a, r(b, 28)) ? 10 : X(a, r(b, 2)) ? r(b, 18) || X(a, r(b, 3)) ? 2 : 0 : !r(b, 18) && X(a, r(b, 3)) ? 1 : -1 : -1
+    }
+
+    function V(a, b) {
+        if (null == b) return null;
+        b = b.toUpperCase();
+        var c = a.a[b];
+        if (null == c) {
+            c = sa[b];
+            if (null == c) return null;
+            c = (new z).a(F.f(), c);
+            a.a[b] = c
+        }
+        return c
+    }
+
+    function X(a, b) {
+        return O(t(b, 3), a) && O(t(b, 2), a)
+    }
+
+    function Sa(a, b) {
+        var c = Ta(a, b);
+        return Ua(a, b, c)
+    }
+
+    function Ua(a, b, c) {
+        var d = t(b, 1),
+            e = T(a, d, c);
+        if (null == e || "001" != c && d != Pa(a, c)) return !1;
+        a = R(b);
+        return -1 != Ra(a, e)
+    }
+
+    function Ta(a, b) {
+        if (null == b) return null;
+        var c = t(b, 1),
+            c = J[c];
```

Comment:
Tagging the places we touch `J`. This one looks safe for multi-language. `t(b, 1)` might be a "lookup with default" function.